### DISCUSSION TITLE
Issue with npmrc with multiple registries

### DISCUSSION THREAD

**Discussion by yaojian-seek** [renovatebot/renovate#11828] [Request Help]
### Context

The app I'm working on uses private packages from 2 npm registries (github, npmjs), 

In my `renovate.json`,  I have 
```
"hostRules": [
  {
    "matchHost": "https://registry.npmjs.org",
    "hostType": "npm",
    "encrypted": {
      "token": "encrypted-npmjs-token"
    }
  },
  {
    "matchHost": "https://npm.pkg.github.com",
    "hostType": "npm",
    "encrypted": {
      "token": "encrypted-github-token"
    }
  }
],
"encrypted": {
  "npmrc": "encrypted-npmrc"
}
```

The unencrypted `npmrc` looks like this, I used `\n` to replace new lines
```
@org:registry=https://npm.pkg.github.com/\n//npm.pkg.github.com/:_authToken=github-token\n//registry.npmjs.org/:_authToken=npmjs-token
```
originally, it looks like this
```
@org:registry=https://npm.pkg.github.com/
//npm.pkg.github.com/:_authToken=github-token
//registry.npmjs.org/:_authToken=npmjs-token
```

### Problem
Renovate app is able to detect dependency updates for the private npm packages hosted in the 2 registries and create Pull Requests for the updates.

But the artifact update fails in the Pull Requests. For example, for the npm package in the github registry, I get this failure message:
```
npm ERR! code E404
npm ERR! 404 Not Found - GET https://npm.pkg.github.com/n/npm.pkg.github.com/:_authToken=**redacted**/n/registry.npmjs.org/:_authToken=**redacted**/@org/package-name
```

It seems like Renovate app doesn't support multiple lines in npmrc with the `\n` or I might be missing something. Any help would be appreciated.

**Comment by viceice** [comment]
Remove all auth from npmrc, renovate will add those from host rules

**Reply by yaojian-seek** [reply]
Hi @viceice , I've tried your suggestion by changing the `npmrc` property to this, without the `encrypted` wrapper
```
"npmrc": "@org:registry=https://npm.pkg.github.com/"
```
It still failed for both registries' private packages with errors indicating the absence of the tokens:
**github**
```
npm ERR! code E401
npm ERR! Unable to authenticate, need: Basic realm="GitHub Package Registry"
```

**npmjs**
```
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/private-package-name - Not found
npm ERR! 404 
npm ERR! 404  'private-package-name@3.2.3' is not in the npm registry.
```

I don't think it's a problem with the `token` I specified in `hostRules`, since Renovate was able to read from the registries and know that there were new updates. Also, the tokens are the same as the ones I use on my local machine to `npm install` successfully.

I've also tried a few other ways that didn't work
 - rearranging the lines in the `npmrc` property
   - private packages from both repositories fail to update package-lock.json
 - having just a single line `//registry.npmjs.org/:_authToken=npmjs-token` in `npmrc`. 
   - this successfully updates the package-lock.json file for the npmjs private packages, but without the 2 other lines, Renovate couldn't do anything with the packages in the github registry

It seems to me that the `\n` is working when performing "Dependency Version Lookups" since it was probably able to interpret `"@org:registry=https://npm.pkg.github.com/"` as a line of its own and perform lookup with the correct registry, as seen in my original question.

But it doesn't work properly when updating artifacts (package-lock.json) as seen with the long line 
```
npm ERR! 404 Not Found - GET https://npm.pkg.github.com/n/npm.pkg.github.com/:_authToken=**redacted**/n/registry.npmjs.org/:_authToken=**redacted**/@org/package-name
```
in the error log in my original question. So, I wonder if there is an existing limitation with multiple lines in the `npmrc` property even with using `\n`.
